From 270f2cab96804e25eabdba25df04fe8bdb240bfe Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Fri, 20 Jul 2018 15:50:28 +0800
Subject: [PATCH 4281/5242] MLK-18721-2: arm64: imx8qm/imx8mm: add touch x/y
 diagonal rotation.

commit  223e4c1435d4b46f610ec41c8165ec894b90f328 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable synaptics touch x/y coordinate diagonal rotation for both
imx8qm and imx8mm evk board.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Reviewed-by: Fugang Duan <fugang.duan@nxp.com>
(cherry picked from commit c6ec617dbc779a7a2d5a68b717995823cd6a3ed8)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts  |    1 +
 .../dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dts  |    1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts
index 06488ae..d7c17e3 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts
@@ -29,6 +29,7 @@
 		pinctrl-0 = <&pinctrl_i2c2_synaptics_dsx_io>;
 		interrupt-parent = <&gpio1>;
 		interrupts = <9 IRQ_TYPE_LEVEL_LOW>;
+		synaptics,diagonal-rotation;
 		status = "okay";
 	};
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dts
index 6aa6e9e..1c58b7d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dts
@@ -132,6 +132,7 @@
 		pinctrl-0 = <&pinctrl_i2c1_synaptics_dsx_io>;
 		interrupt-parent = <&gpio5>;
 		interrupts = <7 8>;
+		synaptics,diagonal-rotation;
 		status = "okay";
 	};
 };
-- 
1.7.9.5

